pandas/_libs/groupby_helper.pxi.in

```diff
@@ -59,24 +76,16 @@ def group_last(rank_t[:, :] out,
             for j in range(K):
                 val = values[i, j]
 
-                # not nan
-                if rank_t is int64_t:
-                    # need a special notna check
-                    if val != NPY_NAT:
-                        nobs[lab, j] += 1
-                        resx[lab, j] = val
-                else:
-                    if val == val:
-                        nobs[lab, j] += 1
-                        resx[lab, j] = val
+                if val == val:
```

Comment:
Is this totally equivalent to what's in place? Looks like we are losing the `NPY_NAT` check?

Reply:
This is in the `rank_t is object` case where we cant use _treat_as_na.  (comment just below here)